### ISSUE TITLE
Add server-side rendering for references

### ISSUE THREAD

**Issue by YaManicKill** [pokemontrades/flairhq#425]
This will make things much quicker, if we do this, and takes us closer to supporting non-JS.

Need to make sure that we can nicely switch to using angular for the references, if we can use javascript. We don't want it to jump around when we do that.

**Comment by YaManicKill** [comment]
Idea:

Block of code that renders the references server-side, but has an ng-if to hide it when we have downloaded the data from the server.

Block of code that renders the references client-side but is hidden until we have downloaded the data from the server.

Pros:

1. References render for non-JS users
2. Page loads quicker for JS users (as we get the html with the references in it initially)
3. Angular takes over when it has downloaded the data, so it can handle adding new references to the page (no dom manipulation)

Cons:

More code. I'm not sure this is a big one, as it can be done nicely. I already need to massively overhaul how we do references as it is, as we have a stupidly long html file that could probably be scaled down quite a bit, and adding in a small amount of duplication isn't a massive issue, tbh.
